backend/repository/handler.go

```diff
@@ -347,6 +348,45 @@ func (h *Handler) CreateDiff(w http.ResponseWriter, r *http.Request, ps httprout
 	h.WriteJSON(w, http.StatusOK, &createDiffResponse{DiffURI: resp})
 }
 
+// LandDiff lands a diff.
+func (h *Handler) LandDiff(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	type LandDiffRequest struct {
```

Comment:
parsing is one-time, but per request. This particular handler is probably of low volume. I wonder if this practice cause overhead for high volume requests, in which case the same type declaration repeats more often.

Regarding readability, I think it makes sense for small structs, and gets awkward for non trivial ones.